PROProtocol/GameClient.cs

```diff
@@ -747,10 +758,12 @@ public bool TakeItemFromPokemon(int pokemonUid)
 
         public bool HasSurfAbility()
         {
-            return HasMove("Surf") &&
+            return (HasMove("Surf") &&
                 (Map.Region == "1" && HasItemName("Soul Badge") ||
                 Map.Region == "2" && HasItemName("Fog Badge") ||
-                Map.Region == "3" && HasItemName("Balance Badge"));
+                Map.Region == "3" && HasItemName("Balance Badge") ||
+                Map.Region == "4" && HasItemName("Relic Badge"))) ||
+                WaterMount != null;

```

Comment:
I think we should check for the badges even if we have a water mount. If we move to a new region, we might not have the required badge yet.
`(HasMove("Surf") || WaterMount != null) && (badge check)`

Reply:
Agreed, assuming PRO's client requires the appropriate badge even with surf mounts. (I have no idea if it does or not)

Reply:
Yes, the client checks if you have the surf badge for surf mounts as well.